apps/explorer/lib/explorer/smart_contract/helper.ex

```diff
@@ -241,15 +241,22 @@ defmodule Explorer.SmartContract.Helper do
   def prepare_license_type(_), do: nil
 
   @doc """
-  Pre-fetches implementation for unverified smart contract or verified proxy smart-contract
+  Pre-fetches implementation for unverified smart-contract or verified proxy smart-contract
   """
   @spec pre_fetch_implementations(Address.t()) :: Implementation.t() | nil
   def pre_fetch_implementations(address) do
     implementation =
       with {:verified_smart_contract, %SmartContract{}} <- {:verified_smart_contract, address.smart_contract},
-           {:proxy?, true} <- {:proxy?, address_is_proxy?(address, @api_true)} do
-        Implementation.get_implementation(address.smart_contract, @api_true)
+           {:proxy?, true} <- {:proxy?, address_is_proxy?(address, @api_true)},
+           # we should fetch implementations only for original smart-contract and exclude fetching implementations of bytecode twin
+           {:bytecode_twin?, false} <- {:bytecode_twin?, address.hash == address.smart_contract.address_hash} do
+        if address.hash == address.smart_contract.address_hash do
+          Implementation.get_implementation(address.smart_contract, @api_true)
+        end
       else
+        {:bytecode_twin?, true} ->
+          Implementation.get_implementation(address.smart_contract, @api_true)
```

Comment:
Maybe I'm wrong, but I expected something like this:

```suggestion
           {:bytecode_twin?, false} <- {:bytecode_twin?, address.hash != address.smart_contract.address_hash} do
        Implementation.get_implementation(address.smart_contract, @api_true)
      else
        {:bytecode_twin?, true} -> nil
```

Reply:
Ouch, I mistakenly notified you here. The final version is applied here https://github.com/blockscout/blockscout/pull/12076/commits/a8f4b8e9ccac9a4c67ca2c7aa54ef145f5fd6e73.